pkg/tekton/metapipeline/client.go

```diff
@@ -0,0 +1,24 @@
+package metapipeline
+
+import (
+	"github.com/jenkins-x/jx/pkg/kube"
+	"github.com/jenkins-x/jx/pkg/tekton"
+)
+
+// Client defines the interface for meta pipeline creation and application.
+type Client interface {
+	// Create creates the Tekton CRDs needed for executing the pipeline as defined by the input parameters
+	Create(pullRef PullRef, pipelineType PipelineKind, context string, envs map[string]string, labels map[string]string) (kube.PromoteStepActivityKey, tekton.CRDWrapper, error)
+
+	// Apply takes the given CRDs for processing, usually applying them to the cluster.
+	Apply(pipelineActivity kube.PromoteStepActivityKey, crds tekton.CRDWrapper) error
+
+	// Close cleans up the resources use by this client.
+	Close() error
+
+	//DefaultImage returns the default image used for pipeline tasks if no other image is specified.
+	DefaultImage() string
+
+	// ServiceAccount returns the service account under which to execute the pipeline.
+	ServiceAccount() string
```

Comment:
Consumer of this pipeline might want to control which `service account` and `namespace` is used. I would make this available in the create `options`.

Reply:
My thinking was that even though we need the service account and default image, they are more client specific options, hence I did not want to add it to single invocation of _Create_. If we introduce an options struct for _Create_ this might be different. I'd probably add them there.